importers/algod/algod_importer_test.go

```diff
@@ -0,0 +1,120 @@
+package algodimporter
+
+import (
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"path"
+	"strconv"
+	"testing"
+
+	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/data/bookkeeping"
+	"github.com/algorand/go-algorand/protocol"
+	"github.com/algorand/go-algorand/rpcs"
+	"github.com/algorand/go-codec/codec"
+	"github.com/algorand/indexer/importers"
+	"github.com/algorand/indexer/plugins"
+	"github.com/sirupsen/logrus"
+	"github.com/sirupsen/logrus/hooks/test"
+	"github.com/stretchr/testify/assert"
+	"gopkg.in/yaml.v3"
+)
+
+var (
+	logger       *logrus.Logger
+	ctx          context.Context
+	cancel       context.CancelFunc
+	s            plugins.PluginConfig
+	testImporter importers.Importer
+)
+
+func init() {
+	logger, _ = test.NewNullLogger()
+	ctx, cancel = context.WithCancel(context.Background())
+	s = ""
+}
+
+func MockAlgodServerReturnsEmptyBlock() *httptest.Server {
```

Comment:
We have somewhat similar code in https://github.com/algorand/indexer/blob/develop/fetcher/fetcher_test.go#L103

Also, Will mentioned that block-generator might be doing the same. We should look to refactor these into commonly used conventions for mocking Algod in the future.

Reply:
I agree. Also `MockAlgodServerReturnsEmptyBlock()` is intentionally exported for other plugins to utilize in testing.